src/query/query_parser/query_parser.rs

```diff
@@ -456,12 +468,12 @@ fn convert_literal_to_query(logical_literal: LogicalLiteral) -> Box<Query> {
     }
 }
 
-fn convert_to_query(logical_ast: LogicalAST) -> Box<Query> {
+fn convert_to_query(logical_ast: LogicalAST, should_match_all: bool) -> Box<Query> {
```

Comment:
`should_match_all` should be renamed `should_match_all_if_empty`.

Reply:
No problem